test/azul_test_case.py

```diff
@@ -20,3 +60,86 @@ def setUpClass(cls) -> None:
     def tearDownClass(cls) -> None:
         boto3.session.Session.get_credentials = cls.get_credentials_orig
         super().tearDownClass()
+
+
+class Test:
+    # Keep this test case out of the main namespace
+
+    class LoggingTestCase(TestCase):
+        """A test case which logs its calls."""
```

Comment:
```suggestion
        """A test case which traces its calls."""
```

Reply:
I'm ok with leaving this in but just know that `mock` does this magically.

See https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock.method_calls

Reply:
I think I will leave it as it is since this works, and I can avoid the complications of having to integrate a Mock class into a TestCase.